### DISCUSSION TITLE
thematicBreak相关问题

### DISCUSSION THREAD

**Discussion by 204065248** [Milkdown/milkdown#812] [Q&A]
请问，https://github.com/Saul-Mirone/milkdown/discussions/331这个链接中的thematicBreak这个代表什么意思呢

**Comment by Saul-Mirone** [comment]
Markdown:
```md

---

```

Rendered:

---

**Reply by 204065248** [reply]
哦哦，那么请问在toMarkdown中AddNode的type如何填写可以加入div之类的东西呢，我使用thematicBreak后他会自动执行到toDOM，让我完成了这件事情

**Reply by Saul-Mirone** [reply]
What feature do you want to implement?

**Reply by 204065248** [reply]
就是通过inputRules实现自定义指令，然后指令会执行toMarkdown。这里我想生成一个html元素。AddNode不知道如何解决这个需求。例如`![]()`会生成下图一样
![image](https://user-images.githubusercontent.com/17307843/201311577-7c4ea3ad-7800-431d-b8f6-1008bb8a6b37.png)

**Reply by 204065248** [reply]
另外我想问问，在inputRules中，使用查询函数查询表情的时候，该查询函数返回值是Promise，如何解决Promise的结果呢，貌似InputRules不支持async await

**Reply by Saul-Mirone** [reply]
1. What you need is not `toMarkdown`. You need to trigger a `prosemirror command` to insert prosemirror node. I sugguest you to take a look at the prosemirror documents.

2. You can insert a placeholder first and replace it with the result. You can view the example in @milkdown/plugin-upload.

**Reply by 204065248** [reply]
您的意思是，在toMarkdown调用command方法吗

**Reply by 204065248** [reply]
在toMarkdown中，我不知道怎么调用command命令😥

**Reply by Saul-Mirone** [reply]
You don't need to do that in toMarkdown. Input rule is triggered by input.

**Reply by 204065248** [reply]
老实说，我有点绕了。我利用inputRules触发规则，再查询我所需要的内容。但查询接口是Promise的结果，以至于我没办法在inputRules中得到结果并渲染到页面中。通过issues中我查阅到支持自定义语法需要remark，但是remark怎么用还不是很明白。当我不使用remark的时候parseMarkdown貌似不会触发，parseDOM也不知道什么时候会触发。😂太难了
